src/lib/telemetry/TelemetryCollector.ts

```diff
@@ -58,8 +66,15 @@ export class TelemetryCollector {
       errors: this.errors.map(err => err.toObject()),
       metrics: this.metrics,
     }
+    const objFilePath = `${TelemetryCollector.telemetryObjFilePathPrefix}-${randomBytes(8).toString('hex')}.json`
+    try {
+      await ensureFile(objFilePath)
+      await writeJson(objFilePath, obj) // Telemetry object should be saved in a file since it can be too large to be passed as a cli argument
+    } catch (e) {
+      console.log('Error writing telemetry file. Error: ', e)
+    }
 
-    spawn(process.execPath, [join(__dirname, 'TelemetryReporter.js'), this.store.storeName, JSON.stringify(obj)], {
+    spawn(process.execPath, [join(__dirname, 'TelemetryReporter.js'), this.store.storeName, objFilePath], {
```

Comment:
I think this has to be inside the `try` just above, or we could `return` inside the `catch`, because if we didn't succeeded writing the file we shouldn't `spawn` the process